repo/blob/azure/azure_storage.go

```diff
@@ -142,6 +150,19 @@ func (az *azStorage) PutBlob(ctx context.Context, b blob.ID, data blob.Bytes, op
 		return translateError(err)
 	}
 
+	if opts.RetentionPeriod != 0 {
+		// it will fail if the retentionPeriod set by the user is lower than that on the policy.
+		retainUntilDate := clock.Now().Add(opts.RetentionPeriod).UTC()
+
+		_, err = az.service.ServiceClient().
+			NewContainerClient(az.Container).
+			NewBlobClient(az.getObjectNameString(b)).
+			SetImmutabilityPolicy(ctx, retainUntilDate, &azblobblob.SetImmutabilityPolicyOptions{})
```

Comment:
Currently it uses the default on the policy rather than setting to Locked/Unlocked based on what the user said. Could be added if preferred. Also I only added Locked as a valid option but I can add both if preferred even though Locked is recommended